### ISSUE TITLE
macOS 13 Ventura Support

### ISSUE THREAD

**Issue by UHDbits** [UHDbits/M73-Tiny-OpenCore#1]
The current version of this configuration does not currently work with macOS 13 Ventura. But, I have gotten Ventura to boot on this ThinkCentre, and you can test it yourself with the "ventura" branch! Currently, beta graphics acceleration is supplied by the latest commit of OpenCore Legacy Patcher. A guide on how to get graphics acceleration working is in the "ventura" branch, but please do note, graphics acceleration is in beta stages. Although I have been using it as my daily for the past couple of months, and have had almost no issues, it is still possible for bugs to appear. Because of this reason, you should wait for the "main" branch of this configuration to support Ventura before using it on this ThinkCentre daily.

**Comment by UHDbits** [comment]
Currently, macOS 13 Ventura doesn't not work on ANY Hackintosh. I tried booting it on the ThinkCentre, and all I got was to the Apple logo (after changing the SMBIOS). I will provide further updates as the situation changes.

**Comment by UHDbits** [comment]
Good news, people at the OpenCore Legacy Patcher community have gotten Ventura to boot on Haswell Macs, although, with graphics acceleration. This is good news for our little ThinkCentre here. Although, no hackintoshs have been known to boot into Ventura yet, so it's still not possible to boot into Ventura.

**Comment by UHDbits** [comment]
People have now gotten further on trying to boot Ventura on a Hackintosh. We have now gotten past an early stage kernel panic, but it still doesn't finish setup. I will be implementing these changes into a new "ventura" branch and I will attempt to commit all newly found changes that may help Ventura boot into this branch.

**Comment by UHDbits** [comment]
Apparently, some people have gotten Ventura to boot on a Hackintosh! I have added all the changes they said to the "ventura" branch and I will be trying it now!

**Comment by UHDbits** [comment]
The config does not boot at all. Do not use it.

**Comment by UHDbits** [comment]
So it seems like "AvoidRuntimeDefrag" being disabled is causing it to not boot.
Which is an issue because I need it to boot Ventura. The current Ventura config is a mess, so I'll remake it from scratch and upload it and see if it works!

**Comment by UHDbits** [comment]
As the day comes to a close, we have gotten a confirmed case of someone running macOS 13 Ventura on a Hackintosh! Sadly, it's not an easy job. And, of course, even if I got it working on this ThinkCentre, we don't have any graphics acceleration. I'll still keep on working to get the config to work with Ventura, maybe even without graphical acceleration.

The new Ventura version of the config will be made completely from scratch to make sure it'll work completely fine without any issues. It'll first be rolled into the "ventura" branch once I can confirm that it is actually bootable. Then, after graphics acceleration patches become available, it'll be rolled into the main branch. SIP will have to be disabled to run Ventura with graphics acceleration, so it'll be disabled by default on the new config.

I will attempt to get this work for everybody as soon as possible!

**Comment by UHDbits** [comment]
New OpenCore versions have been released to help try to get Ventura working more easily on more machines. I'll try again to get Ventura working today, but I will not be uploading any new OpenCore versions to the Ventura repo for the time being. This will just make it quicker for me to go ahead and test the config.

**Comment by UHDbits** [comment]
Yeah, it's not happening. I'll try again when the "AvoidRuntimeDefrag" quirk can be enabled again, because that seems to be the main issue. The "ventura" branch has been reset to be identical to "main". I may just delete the EFI folder in the "ventura" branch and put in the README that the EFI will becoming soon. Again, the new EFI will be made completely from scratch.

**Comment by UHDbits** [comment]
Ok, so I'm going to try to create a new OpenCore config that supports Ventura the week after next week. If it doesn't work, I'll basically try to create a new config with updated OpenCore and kexts every week.

**Comment by UHDbits** [comment]
Amazing news! I have gotten the Ventura installer booted using a brand new OpenCore config! I am currently installing it to make sure that it'll continue working. After I install it, I will immediately upload the config to the "ventura" branch. Again, this is a debug config with debug features enabled, and there is no graphics acceleration. This makes Ventura basically unusable. This thread will now be dedicated to updates on graphics acceleration patches for Ventura. Tommorow, I will update the config I upload to a more finalized form, without the debug features and with Ethernet drivers and all those other things.

**Comment by UHDbits** [comment]
And, it works! No graphics acceleration so it's very slow, but it works! I'll upload the debug config to the "ventura" branch ASAP, and I'll start work on the more finalized version tomorrow. (it says iMac 2015 because I accidentally put the serial number of a 2015 iMac. Whoops. I will fix that in the finalized version.) This thread will now be used to track progress of getting graphics acceleration working on Haswell.
![Ventura Screenshot](https://user-images.githubusercontent.com/66931349/174525319-21635d9f-e424-4b80-b577-4be1148a7b9c.png)

**Comment by UHDbits** [comment]
If you do want to run macOS Ventura on your ThinkCentre, I recommend using Parallels or VMware. Even with an unsupported GPU, you get acceleration in Ventura!
![Screen_Shot_2022-07-16_at_12.png](https://user-images.githubusercontent.com/66931349/179370858-006d92bf-b289-478b-9122-88816c1cbd40.png)

**Comment by UHDbits** [comment]
There has been no updates on graphics acceleration for unsupported GPUs on macOS Ventura from the OpenCore Legacy Patcher team. It is still recommended to use a virtual machine software like VMWare or Parallels to run Ventura for the time being.

**Comment by UHDbits** [comment]
Great news, the AMAZING OpenCore Legacy Patcher developers have gotten preliminary Metal acceleration on many legacy Metal GPUs as shown in [this tweet](https://twitter.com/khronokernel/status/1561602826534457346)! I do not yet know if it works on Haswell iGPUs, but I will test in a small bit. Note, although graphics acceleration has been gained (again, not sure if it works on Haswell, but I will test in just a small bit), it does NOT mean that the "main" branch will support Ventura yet. This is preliminary graphics acceleration, which means it may be buggy. Once OCLP 0.5.0 has been released, I will update the main branch to support Ventura, and close this issue. Anyways, I am going to test Haswell graphics acceleration now, and I will update you ASAP if it does work.

**Comment by UHDbits** [comment]
Apparently the "ventura-alpha" branch of OCLP DOES support Haswell graphics acceleration! I am currently flashing a Ventura USB right now and I will show the results here once it is done.

**Comment by UHDbits** [comment]
After root-patching, I seem to be unable to successfully boot into Ventura. Obviously again, these patches are preliminary and are technically mainly meant to be ran on real Macs. I am going to continue trying for bit, and if it still doesn't work I'll wait for more progress to be made in this front.

**Comment by UHDbits** [comment]
It seems like the reason why I was not able to boot was that my csr_active_config value was not high enough. The required value was recently added in a new OCLP commit. I was having AMFI errors, and the new commit says these SIP values are required to disable AMFI, so I am fairly sure that is the reason why I was not able to boot Ventura after root patching. I will try again today or tomorrow and see if I can get graphics acceleration. I am thinking if I will include the SIP value in the config.plist, or make people set it manually, as I would prefer to have SIP enabled out of the box. Or, I will set up the config to support Ventura, and create a separate guide on how to increase the security of your OpenCore config. I will probably do the Security guide, so people that actually do care can follow the guide. The guide will not work on Ventura.

**Comment by UHDbits** [comment]
![Screenshot 2022-08-23 at 3 17 30 PM](https://user-images.githubusercontent.com/66931349/186279089-30b08038-d656-4949-b2a1-1f3d7dc328dc.png)
I have gotten Ventura running with graphics acceleration on the ThinkCentre! It's not an easy process, and requires disabling some security features. I'll likely have those security features disabled out of the box on this OpenCore config, and give a guide on how to increase security. I'll start working on the two guides in a couple of days.

**Comment by UHDbits** [comment]
Going to start on the new documentation today. I need to reinstall Ventura first and get some other work done, but I hope to at least get most of the documentation done by the end of the weekend. It may not be done though, but I will provide updates here as work is done on it.

**Comment by UHDbits** [comment]
Forgot to post this, but the guide is basically done for how to use OpenCore Legacy Patcher. Now, I just mainly need to create a guide on how to improve security, and then the guides will be done! Once OpenCore Legacy Patcher 0.5.0 is actually released, these changes will be merged into the "main" branch, and the "ventura" branch will be deleted.

**Comment by UHDbits** [comment]
Been a while, but not much has happened on the "ventura" branch. I hope to actually get it basically finished this week, but I may have to make a couple of updates to the configuration before being completely done. I'll likely update the config first and then update the documentation, but once those are done I may create an "alpha" release that does support Ventura, so it's more clear to find.

**Comment by UHDbits** [comment]
Hey everyone, it's time for another update!

I've been using Ventura on this ThinkCentre for quite a while at this point, and it's been surprisingly stable. Especially with the recent release of Ventura Beta 9, which fixed many issues I had. Sadly, OpenCore Legacy Patcher 0.5.0 still does not have a stable release, and it is not expected to have one for a while. I do not want to release a stable version of the config for Ventura until OCLP has been updated for Ventura. This will hopefully be in the next couple of months, but there is no schedule at the moment.

**Comment by UHDbits** [comment]
OpenCore Legacy Patcher 0.5.0 is now available for beta testing, so I have updated the OCLP instructions according to the new links. This hopefully means OCLP 0.5.0 will be released soon, and after a couple of final changes, "ventura-alpha" will be merged into "main". 
As an update on how Ventura has been running on this ThinkCentre, well, it's been amazing. Especially with the recent Ventura Release Candidate, it's been as fast if not faster then Monterey. I'm excited to merge the changes made for Ventura into main, hopefully soon!

**Comment by UHDbits** [comment]
macOS 13 Ventura has been released! OCLP 0.5.0 is likely going to release tomorrow, so I'm hoping to get the "ventura" branch updated in time for tomorrow. In the event where that does not happen, the "ventura" branch will be merged into "main" on Wednesday. Thank you for your patience during this time!

**Comment by UHDbits** [comment]
After many months of waiting, both macOS 13 Ventura and OpenCore Legacy Patcher have been released! This means that support for Ventura is finally stable and can be merged into main.

There are still some final documentation changes I have to make, so Ventura support will be fully released with the release of OpenCore 0.8.6, but I now consider Ventura stable enough to be used as a daily. Thank you for your patience during this time, now let's just hope that macOS 14 doesn't completely discontinue this ThinkCentre.
